src/js/client/editprefset-dialogs-widget.js

```diff
@@ -0,0 +1,339 @@
+/**
+ * Dialog boxes
+ *
+ * - addContextDialog Adds a preference set to a preferences safe
+ * - editContextDialog Edits metadata for a preference set
+ * - confirmDeleteContextDialog Confirms with the user before permanently
+ *   removing a preference set from a safe
+ * - confirmAddProductDialog After clicking a product to add to
+ *   preference safe, confirms before actual insertion.
+ * - confirmSaveDialog Shows a list of changes since last save, and
+ *   confirms with the user before saving preference set.
+ * - confirmRemoveProductDialog Confirms with the user before permanently
+ *   removing a product from the preference safe.
+ *
+ * Copyright 2019 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.dialogs");
+
+/**
+ * Add NP Set Context Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.addContextDialog", {
```

Comment:
All components in this file are heavily bounded with the parent component `gpii.devpmt.editPrefs`. It would be nice to decouple the implementation using model relay and other methods.